app/frontend/src/components/designer/FormViewer.vue

```diff
@@ -354,27 +339,69 @@ export default {
           await new Promise((resolve) => setTimeout(resolve, 500));
         }
         if (this.confirmSubmit) {
-          this.doSubmit(submission, next);
+          this.confirmSubmit = false; // clear for next attempt
+          next();
+        } else {
+          // Force re-render form.io to reset submit button state
+          this.reRenderFormIo += 1;
         }
-        // Force re-render form.io to reset submit button state
-        this.reRenderFormIo += 1;
       } else {
-        this.doSubmit(submission, next);
+        next();
       }
     },
 
+    // FormIO submit event
     // eslint-disable-next-line no-unused-vars
     async onSubmit(submission) {
       if (this.preview) {
         alert('Submission disabled during form preview');
         return;
       }
+
+      const errors = await this.doSubmit(submission);
+
       // if we are here, the submission has been saved to our db
       // the passed in submission is the formio submission, not our db persisted submission record...
       // fire off the submitDone event.
       // console.info(`onSubmit(${JSON.stringify(submission)})`) ; // eslint-disable-line no-console
-      this.currentForm.events.emit('formio.submitDone');
+      if (errors) {
+        this.addNotification({
+          message: errors,
+          consoleError: `Error submiting the form: ${errors}`,
+        });
+      } else {
+        this.currentForm.events.emit('formio.submitDone');
+      }
+    },
+
+    // Not a formIO event, our saving routine to POST the submission to our API
+    async doSubmit(submission) {
+      // console.info(`doSubmit(${JSON.stringify(submission)})`) ; // eslint-disable-line no-console
+      // since we are not using formio api
+      // we should do the actual submit here, and return any error that occurrs to handle in the submit event
+      let errMsg = undefined;
+      try {
+        const response = await this.sendSubmission(false, submission);
+
+        if ([200, 201].includes(response.status)) {
+          // all is good, flag no errors and carry on...
+          // store our submission result...
+          this.submissionRecord = Object.assign(
+            {},
+            this.submissionId ? response.data.submission : response.data
+          );
+          // console.info(`doSubmit:submissionRecord = ${JSON.stringify(this.submissionRecord)}`) ; // eslint-disable-line no-console
+        } else {
+          // console.error(response); // eslint-disable-line no-console
+          errMsg = 'An error occurred submitting this form';
```

Comment:
This else block should throw an error instead of repeating the same errMsg assignment - then it'd be using the same catch block logic.

Reply:
Indeed 👍